### ISSUE TITLE
Illegal Reflective Access Warning with Java 9+

### ISSUE THREAD

**Issue by lukaseckert** [cloudyrock/mongock#68]
**Info**
- mongock-core 2.0.0
- mongock-spring 2.0.0
- Java(TM) SE Runtime Environment 18.9 (build 11.0.3+12-LTS)

**Description**
After upgrading our project from JDK8 to JDK11, the JVM prints an illegal reflective access warning when Mongock is initialized:
```
WARNING: Illegal reflective access by net.sf.cglib.core.ReflectUtils$1 (file:/C:/Users/USER/.gradle/caches/modules-2/files-2.1/cglib/cglib/3.2.12/16c0d1d8b5d50ea9ad38c1f6f9f1e35a42727bf0/cglib-3.2.12.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
	at net.sf.cglib.core.ReflectUtils$1.run(ReflectUtils.java:61)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at net.sf.cglib.core.ReflectUtils.<clinit>(ReflectUtils.java:52)
	at net.sf.cglib.core.KeyFactory$Generator.generateClass(KeyFactory.java:243)
	at net.sf.cglib.core.DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25)
	at net.sf.cglib.core.AbstractClassGenerator.generate(AbstractClassGenerator.java:332)
	at net.sf.cglib.core.AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:96)
	at net.sf.cglib.core.AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:94)
	at net.sf.cglib.core.internal.LoadingCache$2.call(LoadingCache.java:54)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at net.sf.cglib.core.internal.LoadingCache.createEntry(LoadingCache.java:61)
	at net.sf.cglib.core.internal.LoadingCache.get(LoadingCache.java:34)
	at net.sf.cglib.core.AbstractClassGenerator$ClassLoaderData.get(AbstractClassGenerator.java:119)
	at net.sf.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:294)
	at net.sf.cglib.core.KeyFactory$Generator.create(KeyFactory.java:221)
	at net.sf.cglib.core.KeyFactory.create(KeyFactory.java:174)
	at net.sf.cglib.core.KeyFactory.create(KeyFactory.java:153)
	at net.sf.cglib.proxy.Enhancer.<clinit>(Enhancer.java:73)
	at com.github.cloudyrock.mongock.ProxyFactory.createProxyClass(ProxyFactory.java:59)
	at com.github.cloudyrock.mongock.ProxyFactory.createProxyFromOriginal(ProxyFactory.java:105)
	at com.github.cloudyrock.mongock.SpringMongockBuilder.build(SpringMongockBuilder.java:98)
	at com.github.cloudyrock.mongock.SpringMongockBuilder.build(SpringMongockBuilder.java:13)

WARNING: Please consider reporting this to the maintainers of net.sf.cglib.core.ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```
At the moment this doesn't seem to be too critical, however I'm concerned about the last statement (future Java releases will deny that kind of operation).

**Reproduce**
I've prepared a small sample project which reproduces the issue:
https://github.com/lukaseckert/mongock_reflectaccess
Make sure to set JAVA_HOME to JDK11, then run
./gradlew bootRun

**Further information**
Other projects seem to have experienced the same issue in the past, see
- For Spring Boot: https://github.com/spring-projects/spring-framework/issues/20414
- For Spring Data: https://jira.spring.io/browse/DATACMNS-1373
- For Gradle: https://github.com/gradle/gradle/issues/3118

**Comment by dieppa** [comment]
Hi @lukaseckert ,

thanks for this. As you say, it's not critical at the moment, but will produces problems with future JDK versions. 

Will fix prioritise this accordingly

**Comment by dieppa** [comment]
This will probably mean that Mongock needs to move from cglib.

**Comment by dieppa** [comment]
Hi @lukaseckert ,

just for your information, we have released 3.0.0.BETA version. We have removed references to cglib in production code. Still using reflections which will be remove in the next days and maybe can produce some warnings. 
The last thing we will need to handle is Mockito, as the version we are using is having these issues, but it will be fix in the next releases
